### PR TITLE
ACM-12171: Add whats new and deprecated info

### PR DESCRIPTION
**Pull request by xenolinux**
A separate what new PR for https://issues.redhat.com/browse/ACM-12171

### CODE REVIEW

File: clusters/release_notes/deprecate_remove.adoc [hunk 1]
```diff
@@ -22,12 +22,25 @@ Learn when parts of the product are deprecated or removed from {mce-short}. Cons
...
 | The `v1alpha1` API is upgraded to `v1beta1` because `v1alpha1` is deprecated.
-| 2.4 
+| 2.4
 | Use `v1beta1`.
 | None
 
 |===
 
+[#hcp-deprecation-aws]
+== The --aws-creds flag deprecation
```
Comment (oafischer) [inline clusters/release_notes/deprecate_remove.adoc]: This needs to be part of the table above, not a new table. The table contents look good though

Comment (xenolinux) [reply clusters/release_notes/deprecate_remove.adoc]: Fixed

File: clusters/release_notes/whats_new.adoc [hunk 2]
```diff
@@ -42,5 +42,7 @@ Learn about new features and enhancements for Cluster lifecycle with {mce-short}
 [#hosted-control-plane]
 == Hosted control planes
 
+* The `--aws-creds` flag in the `hcp` command line interface (CLI) is deprecated. Now, you need to use the `--sts-creds` and `--role-arn` flags in the `hcp` CLI. To use the `--sts-creds` and `--role-arn` flags, you must create an AWS IAM role and STS credentials. For more information, see xref:../hosted_control_planes/create_role_sts_aws.adoc#create-role-sts-aws[Creating an AWS IAM role and STS credentials].
```
Comment (oafischer) [inline clusters/release_notes/whats_new.adoc]: We do not announce deprecations in what's new, it's only for new features.

Comment (xenolinux) [reply clusters/release_notes/whats_new.adoc]: Ack. Rephrased

File: clusters/release_notes/whats_new.adoc [hunk 3]
```diff
@@ -42,5 +42,7 @@ Learn about new features and enhancements for Cluster lifecycle with {mce-short}
 [#hosted-control-plane]
 == Hosted control planes
 
+* Now, you need to use the `--sts-creds` and `--role-arn` flags instead of the `--aws-creds` flag in the `hcp` command line interface. To use the `--sts-creds` and `--role-arn` flags, you must create an Amazon Web Services (AWS) Identity and Access Management (IAM) role and Security Token Service (STS) credentials. For more information, see xref:../hosted_control_planes/create_role_sts_aws.adoc#create-role-sts-aws[Creating an AWS IAM role and STS credentials].
```
Comment (oafischer) [inline clusters/release_notes/whats_new.adoc]:
```suggestion
* The `--sts-creds` and `--role-arn` flags replace the deprecated `--aws-creds` flag in the `hcp` command line interface. Create an Amazon Web Services (AWS) Identity and Access Management (IAM) role and Security Token Service (STS) credentials to use the `--sts-creds` and `--role-arn` flags. For more information, see xref:../hosted_control_planes/create_role_sts_aws.adoc#create-role-sts-aws[Creating an AWS IAM role and STS credentials].
```

What's new generally needs to be positive, reworded with Brandi's help

Comment (xenolinux) [reply clusters/release_notes/whats_new.adoc]: Thanks. Addressed.

**Review by oafischer — APPROVED** [review]

**Pull request merged**